src/IO/HDF5/HDF5IOHandler.cpp

```diff
@@ -2184,6 +2208,47 @@ void HDF5IOHandlerImpl::readAttribute(
             status = H5Aread(attr_id, attr_type, vcld.data());
             a = Attribute(vcld);
         }
+        else if (H5Tequal(attr_type, m_H5T_CLONG_DOUBLE))
+        {
+            std::vector<std::complex<long double> > vcld(dims[0], 0);
+            status = H5Aread(attr_id, attr_type, vcld.data());
+            a = Attribute(vcld);
+        }
+        else if (H5Tequal(attr_type, m_H5T_LONG_DOUBLE_80_BE))
+        {
+            // worst case, sizeof(long double) is only 8, so allocate enough
+            // memory to fit 16 bytes per member
+            std::vector<long double> vld80be(dims[0] * 2, 0);
+            status = H5Aread(attr_id, attr_type, vld80be.data());
+            H5Tconvert(
+                attr_type,
+                H5T_NATIVE_LDOUBLE,
+                dims[0],
+                vld80be.data(),
+                nullptr,
+                H5P_DEFAULT);
+            std::cout << attr_name << " size " << dims[0] << ", " << vld80be[0]
+                      << " " << vld80be[1] << " " << vld80be[2] << std::endl;
+            a = Attribute(vld80be);
+        }
+        else if (H5Tequal(attr_type, m_H5T_LONG_DOUBLE_80_LE))
+        {
+            // worst case, sizeof(long double) is only 8, so allocate enough
+            // memory to fit 16 bytes per member
```

Comment:
We should maybe use malloc here to avoid running into aliasing issues?

Reply:
Yep, that makes much more sense. This was just my stop-gap solution for the prototype.